subtests/docker_cli/deferred_deletion/deferred_deletion.py

```diff
@@ -0,0 +1,145 @@
+r"""
+Summary
+---------
+
+Test deferred deletion: device mapper will not delete a docker
+filesystem if it's still in use.
+
+This test is only applicable if docker daemon is run with:
+
+    --storage-opts dm.use_deferred_deletion=true
+
+(this is detected and enabled by the docker-storage-setup script)
+
+That suffices on Fedora. On RHEL it's more complicated: the 7.2 version
+of docker-storage-setup blindly enabled the option even though it
+didn't actually work due to a kernel bug. 7.4 introduced a kernel with
+a fix, but disabled by default. We therefore need to differentiate
+between systems with (1) enabled incorrectly and correctly. We do
+so by checking that the sysctl knob /proc/sys/fs/may_detach_mounts
+(introduced in kernel 3.10.0-632) exists and is 1.
+
+We abort with TestNAError if any precondition fails.
+
+Note that we need to do all the important work from inside a
+helper script, because our autotest process will only have access
+to the container's mounts if MountFlags=slave is *absent* from
+the docker systemd unit file; by default that option is present,
+and it is not clear if/when it will be removed. To make sure
+we have access to the container's mounts we need to nsenter
+the docker daemon's mount namespace, and we can't do that within
+autotest (using ctypes.CDLL and setns) because autotest is
+multithreaded and setns() doesn't allow that.
```

Comment:
A bit much for a summary, eh?  See if you can move the bulk of this under an ``Operational Detail`` heading.

Background:  A long time ago, in a galaxy far, far, far, far away.  We were asked to bundle each test's summary in with it's results.  A lot of work was done to the documentation processing modules to enable doing this.  I'm willing to bet that at some point, having test results w/ summaries for reporting purposes will come again.  However, this will only remain possible if the test summary sections are kept relatively short and summary-ish 😀